test/helpers.lua

```diff
@@ -185,7 +188,11 @@ local function check_cores(app, force)
   local gdb_db_cmd = 'gdb -n -batch -ex "thread apply all bt full" "$_NVIM_TEST_APP" -c "$_NVIM_TEST_CORE"'
   local lldb_db_cmd = 'lldb -Q -o "bt all" -f "$_NVIM_TEST_APP" -c "$_NVIM_TEST_CORE"'
   local random_skip = false
-  local local_tmpdir = tmpdir_is_local(tmpdir_get()) and tmpdir_get() or nil
+  -- Workspace-local $TMPDIR, scrubbed and pattern-escaped.
+  -- "./Xtest-tmpdir/" => "Xtest%-tmpdir"
+  local local_tmpdir = (tmpdir_is_local(tmpdir_get())
+    and tmpdir_get():gsub('^[ ./]+',''):gsub('%/+$',''):gsub('([^%w])', '%%%1')
```

Comment:
> workspace-local temp dir is not going to be an absolute path, and even if it is, removing the preceding slash isn't going to cause a false positive.

I don't see how that's guaranteed.  `$TMPDIR`/`$TEMP` could be absolute and `tmpdir_is_local()` only checks if `Xtest` is in the string, not whether it's a child of the working directory.

---
Ah, `RunTests.cmake` sets `TMPDIR` to adhere to what you're checking.  Might be a good idea to mention that point in the comment you added.